### ISSUE TITLE
`boost::apply_visitor()` no longer allows visitors to accept lvalues (regression)

### ISSUE THREAD

**Issue by ras0219-msft** [boostorg/variant#69]
As part of updating vcpkg to use boost 1.70[1], I keep running into issues around the use of `boost::apply_visitor`. Specifically, it looks like in 1.70, it is no longer valid to visit by lvalue reference:
```
#include <boost/variant.hpp>

struct T{};
struct U{};

struct V {
        void operator()(T&) const {}
        void operator()(U&) const {}
};

int main()
{
    V v;
    boost::variant<T,U> var(U{});

    boost::apply_visitor(v, var);
    return 0;
}
```

The commit that broke this is https://github.com/boostorg/variant/commit/4b37f9b804ee39e96184224447eecc0ad4dc4254; can someone help to confirm if this was intended to break or whether this is an unintended regression?

@Kojoley as the commit author

[1] https://github.com/Microsoft/vcpkg/pull/6141

**Comment by apolukhin** [comment]
Fixed in PR #68

**Comment by yurivict** [comment]
@apolukhin Please make a boost release with this fix. This problem was apparently a regression that broke facebook's ```fuzz``` project. See here: https://github.com/facebookincubator/fizz/issues/33

**Comment by apolukhin** [comment]
The fix will be in Boost 1.71. Until then you can manually apply the patch from https://github.com/boostorg/variant/pull/68 to the 1.70 version

**Comment by agnosticdev** [comment]
Thank you for this patch I was able to apply it to `Boost 1.70` and get the `fizz` project building on Mojave 10.14.5.
